oak_core/src/sync.rs

```diff
@@ -51,6 +51,35 @@ impl<T> OnceCell<T> {
         Some(unsafe { (*self.value.get()).assume_init_ref() })
     }
 
+    /// Gets the contents of the cell, initializing it with `f` if
+    /// the cell was empty. If the cell was empty and `f` failed, an
+    /// error is returned.
+    ///
+    /// Based off the method of the same name in [`core::cell::OnceCell`].
+    ///
+    /// # Panics
+    ///
+    /// If `f` panics, the panic is propagated to the caller, and the cell
+    /// remains uninitialized.
+    ///
+    /// It is an error to reentrantly initialize the cell from `f`. Doing
+    /// so results in a panic.
+    pub fn get_or_init<F>(&self, f: F) -> &T
```

Comment:
I avoided adding this type of function on purpose, since this just turns the OnceCell into a lazy_static. I want to avoid lazy initialisation of things that we will always use, rather than just avoid the lazy_static crate. My preference would be to initialise the dice data explicitly during startup, deal with the result at that time and store the `DiceWrapper` (rather than storing an `anyhow::Result`) in the OnceCell.

Reply:
Fair enough. The reason I went with this route here is that in order to explicitly init the dice data during startup, the SDK would need to expose an init function ala `pub fn init() -> anyhow::Result<()>`. — That would deviate from the SDK interface outlined in the design doc. It would no longer be needed once we have the restricted kernel orchestrator. 

Would you be okay with keeping the `anyhow::Result` in the OnceCell here, but removing `get_or_init` from `oak_core` (instead inlining the logic in the SDK)?

If not I can go ahead with init function, and sync with @ipetr0v about it once he's back in the office.

Reply:
Actually, that is a good point that I forgot about. Avoiding lazy initialisation is more important in the kernel and stage  0. Since the SDK will be used at the application level only, lazy initialization seems fine. I would in fact prefer that you go back to using `lazy_static` in this case rather than enabling lazy initialization on the `OnceCell` implementation. Apolpogies for going backwards and forwards on this.